src/app/components/PriceChart.jsx

```diff
@@ -32,37 +110,65 @@ const priceChartConfig = merge({}, config.highCharts.priceChart, {
       valueDecimals: 2
     },
     threshold: null
-  }]
-})
-
-class PriceChart extends Component {
-  componentDidUpdate (prevProps) {
-    const symbol = this.props.symbol
-    if (!prevProps.chartOpen && this.props.chartOpen) {
-      const priceChart = this.refs[`priceChart_${symbol}`].getChart()
-      //set data to blank so if multiple charts open doesn't show previous data while loading
-      priceChart.series[0].setData([])
-      priceChart.showLoading()
-      fetchPriceChart(symbol)
-        .then((data) => {
-          priceChart.series[0].setData(data)
-          priceChart.hideLoading()
-        })
-        .catch(e => {
-          log.error(e)
-          toastr.error(`Error getting price chart data for ${symbol}`)
-        })
-    }
-  }
-
-  render () {
-    return <ReactHighstock config={priceChartConfig} ref={`priceChart_${this.props.symbol}`} />
+  }],
+  yAxis: {
+    min: 0,
+    opposite: false,
+    gridLineColor: '#707073',
+    title: {
+      text: priceSeriesName,
+      style: {
+        color: themeColor.primary
+      }
+    },
+    labels: {
+      y: null, // Center vertically
+      format: '${value}',
+      align: 'right',
+      style: {
+        color: themeColor.primary,
+        fontSize: '12px'
+      }
+    },
+    lineColor: '#707073',
+    minorGridLineColor: '#505053',
+    tickColor: '#707073',
+    tickWidth: 1,
+    plotLines: [{
+      value: 0,
+      width: 1,
+      color: '#808080'
+    }]
   }
 }
 
-PriceChart.propTypes = {
-  symbol: PropTypes.string.isRequired,
-  chartOpen: PropTypes.bool
-}
-
-export default connect()(PriceChart)
+export default compose(
+  setDisplayName('PriceChart'),
+  connect(createStructuredSelector({
+    priceChartData: (state, { symbol }) => getPriceChartData(state, symbol)
+  }), {
+    fetchPriceChart: fetchPriceChartData
+  }),
+  setPropTypes({
+    symbol: PropTypes.string.isRequired,
+    chartOpen: PropTypes.bool
+  }),
+  defaultProps({
+    chartOpen: false
+  }),
+  withProps(({ priceChartData }) => {
+    const priceChartConfig = initialPriceChartConfig
+    if (priceChartData) {
+      priceChartConfig.series[0].data = priceChartData.priceData
```

Comment:
This modifies the initialPriceChartConfig. Avoid doing this and return a modified copy of the config like so:
```
{
  ...initialPriceChartConfig,
  series: [{
    ...initialPriceChartConfig.series[0],
    data: priceChartData.priceData
  }]
}
```